lib/ansible/modules/cloud/azure/azure_rm_localnetworkgateway.py

```diff
@@ -0,0 +1,246 @@
+
+
+bgp_spec = dict(
+    asn=dict(type='int'),
+    bgp_peering_address=dict(),
+    peer_weight=dict(type='int', default=0)
+)
+
+
+class AzureRMLocalNetworkGateway(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            location=dict(type='str'),
+            local_address_prefixes=dict(type='list', default=[]),
+            gateway_ip_address=dict(),
+            bgp_settings=dict(type='dict', options=bgp_spec),
+        )
+
+        required_if = [
+            ('state', 'present', ['name', 'gateway_ip_address']),
+            ('state', 'absent', ['name'])
+        ]
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.location = None
+        self.gateway_ip_address = None
+        self.local_address_prefixes = None
+        self.bgp_settings = None
+
+        self.results = dict(changed=False)
+
+        super(AzureRMLocalNetworkGateway, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                         supports_check_mode=True,
+                                                         required_if=required_if,
+                                                         supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+        changed = False
+        results = dict()
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        results = self.get_lgw()
+        if not results:
+            changed = True if self.state == 'present' else False
+        else:
+            if self.state == 'present':
+                update_tags, results['tags'] = self.update_tags(results.get('tags', None))
+                if update_tags:
+                    changed = True
+                if results['gateway_ip_address'] != self.gateway_ip_address:
+                    changed = True
+                bgp_settings = self.bgp_settings if self.bgp_settings else None
+                if bgp_settings != results.get('bgp_settings', None):
+                    changed = True
+                if sorted(results['local_network_address_space']['address_prefixes']) != sorted(self.local_address_prefixes):
+                    changed = True
+            else:
+                self.log("CHANGED: local network gateway exists but requested state is 'absent'")
+                changed = True
+
+        self.results['changed'] = changed
+        self.results['id'] = None
+        if self.check_mode:
+            self.results['changed'] = True
```

Comment:
self.results['changed']=changed

Reply:
Please refer earlier comment on integration test for 'check_mode'.